### PR TITLE
Feat/beta sandboxes

### PR DESCRIPTION
**Pull request by danilowoz**

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/5973/commits/40a6903f37bbe14b80fd16a74fe735cc697f5f06 is at https://pr5973.build.csb.dev/s/new.

File: packages/app/src/app/pages/Dashboard/Components/Beta/ImportRepoBetaModal.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    }
+  }, []);
+
+  return (
+    <ThemeProvider>
+      <Modal
+        isOpen={modals.importRepoBeta.isCurrent}
+        onClose={() => modalsActions.importRepoBeta.close()}
+        width={950}
+        fullWidth={window.screen.availWidth < 800}
```
Comment (smhutch) [inline packages/app/src/app/pages/Dashboard/Components/Beta/ImportRepoBetaModal.tsx:46]: Never seen `screen.availWidth` before. TIL! 😅

File: packages/app/src/app/pages/Dashboard/Components/Beta/RepoBetaCard.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  isScrolling,
+  ...props
+}) => {
+  const [stoppedScrolling, setStoppedScrolling] = React.useState(false);
+
+  React.useEffect(() => {
+    if (!isScrolling && !stoppedScrolling) {
+      setStoppedScrolling(true);
+    }
+  }, [isScrolling, stoppedScrolling]);
```
Comment (smhutch) [inline packages/app/src/app/pages/Dashboard/Components/Beta/RepoBetaCard.tsx:22]: Doesn't seem to be used for anything. Could we remove?

File: packages/app/src/app/pages/Dashboard/Components/Beta/RepoBetaCard.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+          <Text
+            size={3}
+            weight="medium"
+            css={css({
+              height: 32,
+              overflow: 'hidden',
+              textOverflow: 'ellipsis',
+              display: '-webkit-box',
+              '-webkit-line-clamp': '2',
+              '-webkit-box-orient': 'vertical',
```
Comment (smhutch) [inline packages/app/src/app/pages/Dashboard/Components/Beta/RepoBetaCard.tsx:80]: What are these for?

File: packages/app/src/app/pages/Dashboard/Components/Beta/RepoBetaListItem.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  isScrolling,
+  ...props
+}) => {
+  const [stoppedScrolling, setStoppedScrolling] = React.useState(false);
+
+  React.useEffect(() => {
+    if (!isScrolling && !stoppedScrolling) {
+      setStoppedScrolling(true);
+    }
+  }, [isScrolling, stoppedScrolling]);
```
Comment (smhutch) [inline packages/app/src/app/pages/Dashboard/Components/Beta/RepoBetaListItem.tsx:24]: Doesn't seem to be used

File: packages/app/src/app/pages/Dashboard/Content/routes/Beta/index.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+  React.useEffect(() => {
+    actions.dashboard.getBetaSandboxes();
+  }, [actions.dashboard]);
+
+  const items: DashboardGridItem[] = sandboxes.BETA
+    ? [
+        { type: 'beta-new-repo' },
+        ...sandboxes.BETA.map(sandbox => ({
+          type: 'beta-repo' as DashboardBetaRepo['type'],
```
Comment (smhutch) [inline packages/app/src/app/pages/Dashboard/Content/routes/Beta/index.tsx:28]: Should work to convert the type from 'string' to `'beta-repo'`:

```suggestion
          type: 'beta-repo' as const,
```

File: packages/app/src/app/pages/Dashboard/types.ts [hunk 6]
```diff
@@ -85,6 +86,25 @@ export type DashboardNewRepo = {
...
+  type: 'beta-new-repo';
+};
+
+type Unpacked<T> = T extends (infer U)[]
+  ? U
+  : T extends (...args: any[]) => infer U
+  ? U
+  : T extends Promise<infer U>
+  ? U
+  : T;
```
Comment (smhutch) [inline packages/app/src/app/pages/Dashboard/types.ts:99]: What does this do? Looks magical 😄

File: packages/app/src/app/pages/Sandbox/index.tsx [hunk 7]
```diff
@@ -142,6 +142,9 @@ export const Sandbox = React.memo<Props>(
         <Helmet>
           <title>{getTitle()} - CodeSandbox</title>
         </Helmet>
+        <div
+          style={{ width: '100%', height: '30px', backgroundColor: 'red' }}
```
Comment (smhutch) [inline packages/app/src/app/pages/Sandbox/index.tsx:146]: Red background intended?

**Review by smhutch — COMMENTED** [review]
A quick skim through the PR this time, since I have already seen the UI working on staging. Just a few small things I spotted :eyes:

File: packages/app/src/app/pages/Dashboard/Content/routes/Beta/index.tsx (hunk 5 above)
Comment (danilowoz) [reply packages/app/src/app/pages/Dashboard/Content/routes/Beta/index.tsx:28]: Good to know!

File: packages/app/src/app/pages/Dashboard/types.ts (hunk 6 above)
Comment (danilowoz) [reply packages/app/src/app/pages/Dashboard/types.ts:99]: It extracts the type of the items from an array, for example
```tsx
Unpacked<string[]> // string
Unpacked<(number | string )[]> // number | string 
```

File: packages/app/src/app/pages/Sandbox/index.tsx (hunk 7 above)
Comment (danilowoz) [reply packages/app/src/app/pages/Sandbox/index.tsx:146]: Definitely this should not be here!

File: packages/app/src/app/pages/Dashboard/Components/Beta/RepoBetaCard.tsx (hunk 3 above)
Comment (danilowoz) [reply packages/app/src/app/pages/Dashboard/Components/Beta/RepoBetaCard.tsx:80]: Good question, it seems it works like a display-flex. As I copied this from the others cards, I'd leave it there as the others components. 
https://www.w3schools.com/cssref/playit.asp?filename=playcss_box-orient

File: packages/app/src/app/components/CreateNewSandbox/CreateSandbox/Import/Import.tsx [hunk 8]
```diff
@@ -103,7 +103,7 @@ export const ImportFromGithub = () => {
           </Stack>
         ) : (
           <PlaceHolderLink error={error}>
-            {error || 'Enter a Github URL and generate a sandbox link'}
+            {error || 'Enter a GitHub URL and generate a sandbox link'}
```
Comment (CompuIves) [inline packages/app/src/app/components/CreateNewSandbox/CreateSandbox/Import/Import.tsx:106]: Very well seen, haha!

**Review by CompuIves — APPROVED** [review]
Great, it hits all the checkmarks of what needs to be done to add a new page. Looks good to me, nice work @danilowoz!

**Pull request merged**